src/steps/data_extraction/utils/dataset_handler.py

```diff
@@ -0,0 +1,277 @@
+    """
+    Retrieves the label map from a dataset version.
+
+    Parameters:
+        dataset_version (DatasetVersion): The dataset version from which to retrieve the label map.
+
+    Returns:
+        dict: A dictionary mapping label names to label objects.
+    """
+    return {label.name: label for label in dataset_version.list_labels()}
+
+
+class DatasetHandler:
+    """
+    Manages dataset versions attached to an experiment and prepares dataset collections for processing.
+
+    This class provides functionality to retrieve dataset versions from an experiment,
+    organize them into dataset contexts based on training, validation, and testing splits,
+    and assemble these contexts into a DatasetCollection for convenient access and use.
+
+    Attributes:
+        experiment (Experiment): The experiment from which datasets are to be retrieved.
+        train_set_split_ratio (float): The proportion of the dataset to be used for training when only one dataset is attached.
+        destination_path (str): The local path where datasets will be stored and accessed.
+    """
+
+    def __init__(self, experiment: Experiment, train_set_split_ratio: float):
+        """
+        Initializes a DatasetHandler with an experiment and configuration for dataset splits.
+
+        Args:
+            experiment (Experiment): The Picsellia Experiment object.
+            train_set_split_ratio (float): The proportion of data to allocate to the training split.
+        """
+        self.experiment = experiment
+        self.train_set_split_ratio = train_set_split_ratio
+        self.destination_path = os.path.join(os.getcwd(), self.experiment.name)
+
+    def get_dataset_collection(self) -> DatasetCollection:
+        """
+        Retrieves dataset versions attached to the experiment and organizes them into a DatasetCollection.
+
+        This method handles different scenarios based on the number of attached datasets: one, two, or three.
+        It prepares dataset contexts for each scenario and assembles them into a DatasetCollection.
+
+        Returns:
+            - DatasetCollection: A collection of dataset contexts prepared based on the attached dataset versions.
+
+        Raises:
+            - ResourceNotFoundError: If the expected dataset splits are not found in the experiment.
+            - RuntimeError: If an invalid number of datasets are attached to the experiment.
+
+        Returns:
+            - DatasetCollection: A collection of dataset contexts prepared based on the attached dataset versions.
+        """
+        nb_attached_datasets = len(self.experiment.list_attached_dataset_versions())
+        try:
+            train_dataset_version = self.experiment.get_dataset(
+                DatasetSplitName.TRAIN.value
+            )
+        except Exception as e:
+            raise ResourceNotFoundError(
+                "Training dataset not found in the experiment."
```

Comment:
Maybe we can improve this message, something like "A training dataset is required and couldn't be found. Please attach a dataset with the alias 'train'."